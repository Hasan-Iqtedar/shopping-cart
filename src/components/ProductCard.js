import React from "react";

const ProductCard = (props) => {
  const { data } = props;

  return (
    <div className="product">
      <img src={data.image} alt={data.title} />
      <div>
        <span id="title">{data.title}</span>
        <span id="price">price: {data.price}</span>
      </div>
      <button>Add to Cart</button>
    </div>
  );
};

export default ProductCard;
