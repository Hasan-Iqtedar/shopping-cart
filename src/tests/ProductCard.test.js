import { render } from "@testing-library/react";
import ProductCard from "../components/ProductCard";

it("Renders a product", () => {
  const product = {
    image: null,
    title: "abc",
    price: 123,
  };

  const { container } = render(<ProductCard data={product} />);

  expect(container).toMatchSnapshot();
});
