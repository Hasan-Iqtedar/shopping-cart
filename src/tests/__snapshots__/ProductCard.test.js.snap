// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a product 1`] = `
<div>
  <div
    class="product"
  >
    <img
      alt="abc"
    />
    <div>
      <span
        id="title"
      >
        abc
      </span>
      <span
        id="price"
      >
        price: 
        123
      </span>
    </div>
    <button>
      Add to Cart
    </button>
  </div>
</div>
`;
